clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java

```diff
@@ -0,0 +1,171 @@
+
+    if (ecsTask != null && instanceStatus != null) {
+      ecsInstance = new EcsInstance(id, ecsTask, instanceStatus);
+    }
+
+    return ecsInstance;
+  }
+
+  @Override
+  public String getConsoleOutput(String account, String region, String id) {
+    return null;
+  }
+
+  private List<String> getAllClusters(AmazonECS amazonECS) {
+    ListClustersResult listClustersResult = amazonECS.listClusters();
+    List<String> clusterList = listClustersResult.getClusterArns();
+    while (listClustersResult.getNextToken() != null) {
+      listClustersResult = amazonECS.listClusters(
+        new ListClustersRequest().withNextToken(listClustersResult.getNextToken())
+      );
+      clusterList.addAll(listClustersResult.getClusterArns());
+    }
+    return clusterList;
+  }
+
+  private NetflixAmazonCredentials getCredentials(String account) {
+    NetflixAmazonCredentials accountCredentials = null;
+    for (AccountCredentials credentials: accountCredentialsProvider.getAll()) {
+      if (credentials.getName().equals(account)) {
+        if (credentials instanceof NetflixAmazonCredentials) {
+          accountCredentials = (NetflixAmazonCredentials) credentials;
+          break;
+        }
+      }
+    }
+
+    if (accountCredentials == null) {
+      throw new NotFoundException(
+        String.format("AWS account %s was not found.  Please specify a valid account name", account)
+      );
+    }
+
+    return accountCredentials;
+  }
+
+  private boolean isValidId(String id, String region) {
+    String id_regex = "[\\da-f]{8}-[\\da-f]{4}-[\\da-f]{4}-[\\da-f]{4}-[\\da-f]{12}";
+    String id_only = String.format("^%s$", id_regex);
+    String arn = String.format("arn:aws:ecs:%s:\\d*:task/%s", region, id_regex);
+    return id.matches(id_only) || id.matches(arn);
+  }
+
+  private Task getTask(AmazonECS amazonECS, String taskId) {
+    Task task = null;
+
+    List<String> queryList = new ArrayList<>();
+    queryList.add(taskId);
+
+    for (String cluster: getAllClusters(amazonECS)) {
+      List<Task> taskList = amazonECS.describeTasks(
+        new DescribeTasksRequest().withCluster(cluster).withTasks(queryList))
+        .getTasks();
+      if (!taskList.isEmpty()) {
+        task = taskList.get(0);
+        break;
+      }
+    }
+    return task;
+  }
+
+
+  private ContainerInstance getContainerInstance(AmazonECS amazonECS, Task task) {
+    if (task == null) {
+      return null;
+    }
+
+    ContainerInstance container = null;
+
+    List<String> queryList = new ArrayList<>();
+    queryList.add(task.getContainerInstanceArn());
+    DescribeContainerInstancesRequest request = new DescribeContainerInstancesRequest()
+      .withCluster(task.getClusterArn())
+      .withContainerInstances(queryList);
+    List<ContainerInstance> containerList = amazonECS.describeContainerInstances(request).getContainerInstances();
+
+    if (!containerList.isEmpty()) {
```

Comment:
From the code, I can guess that you can have more than 1 container instance on an ECS instance...which is a bit insane and runs counter to what the ECS folks have told us regarding their API

That being said, I think it'd be a good thing to add a check for the size of that container list, and throw an error if the size is larger than 1.  That'd be an unexpected behaviour that would deserve being quickly identified and debugged, instead of having unexpected behaviours creep in as hard-to-track bugs

Reply:
I am not sure if you would ever get more than one ECS instance, but the result is always an array of size = 1. Added the checks